src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc

```diff
@@ -266,49 +268,59 @@ void ResetGlobalPropertiesRequest::on_event(const event_engine::Event& event) {
     }
   }
 
-  if (!IsPendingResponseExist()) {
-    bool result =
-        ((hmi_apis::Common_Result::SUCCESS == ui_result_) &&
-         (hmi_apis::Common_Result::SUCCESS == tts_result_ ||
-          hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == tts_result_)) ||
-        ((hmi_apis::Common_Result::SUCCESS == ui_result_) &&
-         (hmi_apis::Common_Result::INVALID_ENUM == tts_result_)) ||
-        ((hmi_apis::Common_Result::INVALID_ENUM == ui_result_) &&
-         (hmi_apis::Common_Result::SUCCESS == tts_result_));
-
-    mobile_apis::Result::eType result_code;
-    const char* return_info = NULL;
-
-    if (result) {
-      if (hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == tts_result_) {
-        result_code = mobile_apis::Result::WARNINGS;
-        return_info =
-            std::string("Unsupported phoneme type sent in a prompt").c_str();
-      } else {
-        result_code = static_cast<mobile_apis::Result::eType>(
-            std::max(ui_result_, tts_result_));
-      }
-    } else {
-      result_code = static_cast<mobile_apis::Result::eType>(
-          std::max(ui_result_, tts_result_));
-    }
+  if (IsPendingResponseExist()) {
+    LOG4CXX_DEBUG(logger_, "Waiting for remaining responses");
+    return;
+  }
 
-    SendResponse(result,
-                 static_cast<mobile_apis::Result::eType>(result_code),
-                 return_info,
-                 &(message[strings::msg_params]));
+  mobile_apis::Result::eType result_code = mobile_apis::Result::INVALID_ENUM;
+  std::string response_info;
+  const bool result = PrepareResponseParameters(result_code, response_info);
 
-    if (!application) {
-      LOG4CXX_ERROR(logger_, "NULL pointer");
-      return;
-    }
+  SendResponse(result,
+               static_cast<mobile_apis::Result::eType>(result_code),
+               response_info.empty() ? NULL : response_info.c_str(),
+               &(message[strings::msg_params]));
 
-    if (result) {
-      application->UpdateHash();
-    }
+  if (!application) {
+    LOG4CXX_ERROR(logger_, "NULL pointer");
+    return;
+  }
+
+  if (result) {
+    application->UpdateHash();
+  }
+}
+
+bool ResetGlobalPropertiesRequest::PrepareResponseParameters(
+    mobile_apis::Result::eType& out_result_code,
+    std::string& out_response_info) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  using namespace helpers;
+
+  bool result = false;
+  ResponseInfo ui_properties_info(ui_result_, HmiInterfaces::HMI_INTERFACE_UI);
+  ResponseInfo tts_properties_info(tts_result_,
+                                   HmiInterfaces::HMI_INTERFACE_TTS);
+
+  if (hmi_apis::Common_Result::SUCCESS == ui_result_ &&
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == tts_result_) {
+    result = true;
```

Comment:
@dcherniev Please add condition that checks state of tts interface. Please clarify, need to change result code if interfase state is "not response"